### PR TITLE
chore: apply clippy suggestions newly introduced in latest compiler

### PR DESCRIPTION
**Pull request by westonpace**

### CODE REVIEW

File: java/core/lance-jni/src/traits.rs [hunk 1]
```diff
@@ -61,10 +61,13 @@ impl FromJString for JString<'_> {
 }
 
 pub trait JMapExt {
+    #[allow(dead_code)]
```
Comment (westonpace) [inline java/core/lance-jni/src/traits.rs:64]: @eddyxu there were other methods here marked with `dead_code` so I am guessing you want to keep these?

File: rust/lance-index/src/vector/graph/builder.rs [hunk 2]
```diff
@@ -172,9 +172,13 @@ impl GraphBuilder {
 
 #[derive(Debug)]
 pub struct GraphBuilderStats {
+    #[allow(dead_code)]
```
Comment (westonpace) [inline rust/lance-index/src/vector/graph/builder.rs:175]: @BubbleCal right now these are only used in tests which the latest clippy considered dead code.  Do we think we might export these someday?  If so, keep it how it is now (with `#[allow(dead_code)]`.  If this is only ever going to be for tests then we can add `#[cfg(test)]` instead.

File: rust/lance-table/src/io/commit.rs [hunk 3]
```diff
@@ -249,9 +245,11 @@ pub trait CommitHandler: Debug + Send + Sync {
 }
 
 /// Adapt an object_store credentials into AWS SDK creds
+#[cfg(dynamodb)]
 #[derive(Debug)]
 struct OSObjectStoreToAwsCredAdaptor(AwsCredentialProvider);
```
Comment (westonpace) [inline rust/lance-table/src/io/commit.rs:250]: This adapter is only needed if we are using dynamodb

**Review by westonpace — COMMENTED** [review]
The large majority of these changes are because rust can now detect a redundant import if you have `use super::*` (e.g. an import is imported in both the `mod test` and the outer implementation)

File: java/core/lance-jni/src/traits.rs (hunk 1 above)
Comment (eddyxu) [reply java/core/lance-jni/src/traits.rs:64]: Yes, let's ignore Java Jni for now

**Review by eddyxu — APPROVED** [review]

**Pull request merged**